diff --git a/hadoop-tools/hadoop-pipes/src/CMakeLists.txt b/hadoop-tools/hadoop-pipes/src/CMakeLists.txt
index ce6ee31..8aaaac0 100644
--- a/hadoop-tools/hadoop-pipes/src/CMakeLists.txt
+++ b/hadoop-tools/hadoop-pipes/src/CMakeLists.txt
@@ -51,19 +51,19 @@ include_directories(
 
 # Example programs
 add_executable(wordcount-simple main/native/examples/impl/wordcount-simple.cc)
-target_link_libraries(wordcount-simple hadooppipes hadooputils)
+target_link_libraries(wordcount-simple hadooppipes hadooputils tirpc)
 hadoop_output_directory(wordcount-simple examples)
 
 add_executable(wordcount-part main/native/examples/impl/wordcount-part.cc)
-target_link_libraries(wordcount-part hadooppipes hadooputils)
+target_link_libraries(wordcount-part hadooppipes hadooputils tirpc)
 hadoop_output_directory(wordcount-part examples)
 
 add_executable(wordcount-nopipe main/native/examples/impl/wordcount-nopipe.cc)
-target_link_libraries(wordcount-nopipe hadooppipes hadooputils)
+target_link_libraries(wordcount-nopipe hadooppipes hadooputils tirpc)
 hadoop_output_directory(wordcount-nopipe examples)
 
 add_executable(pipes-sort main/native/examples/impl/sort.cc)
-target_link_libraries(pipes-sort hadooppipes hadooputils)
+target_link_libraries(pipes-sort hadooppipes hadooputils tirpc)
 hadoop_output_directory(pipes-sort examples)
 
 add_library(hadooputils STATIC
