diff --git a/ambari-common/src/main/python/ambari_commons/os_check.py b/ambari-common/src/main/python/ambari_commons/os_check.py
index e242d7b..2001b2d 100644
--- a/ambari-common/src/main/python/ambari_commons/os_check.py
+++ b/ambari-common/src/main/python/ambari_commons/os_check.py
@@ -104,7 +104,15 @@ def advanced_check(distribution):
       
       if search_groups:
         distribution[1] = search_groups.group(1)
-      
+
+
+    if distribution[0] == '':
+      distribution[0] = name.lower()
+
+      search_groups = re.search('VERSION_ID="(.+)"', file_content)
+      if search_groups:
+        distribution[1] = search_groups.group(1)
+
   return tuple(distribution)
     
 
@@ -254,6 +262,8 @@ class OSCheck:
     # special cases
     if _is_oracle_linux():
       operatingSystem = 'oraclelinux'
+    elif operatingSystem.startswith('opensuse leap'):
+      operatingSystem = 'sles'
     elif operatingSystem.startswith('suse linux enterprise server'):
       operatingSystem = 'sles'
     elif operatingSystem.startswith('red hat enterprise linux'):
diff --git a/ambari-common/src/main/python/ambari_commons/resources/os_family.json b/ambari-common/src/main/python/ambari_commons/resources/os_family.json
index 0634988..3eb1635 100644
--- a/ambari-common/src/main/python/ambari_commons/resources/os_family.json
+++ b/ambari-common/src/main/python/ambari_commons/resources/os_family.json
@@ -71,7 +71,8 @@
         ],
         "versions": [
           11,
-          12
+          12,
+          15
         ]
       },
       "winsrv": {
diff --git a/ambari-server/src/main/resources/version_definition.xsd b/ambari-server/src/main/resources/version_definition.xsd
index fb38370..65ad3af 100644
--- a/ambari-server/src/main/resources/version_definition.xsd
+++ b/ambari-server/src/main/resources/version_definition.xsd
@@ -63,6 +63,7 @@
       <xs:enumeration value="ubuntu18" />
       <xs:enumeration value="suse11" />
       <xs:enumeration value="suse12" />
+      <xs:enumeration value="suse15" />
       <xs:enumeration value="amazonlinux2" />
     </xs:restriction>
   </xs:simpleType>
